spec/support/view_component/example_group.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Support
+  module ViewComponent
+    module ExampleGroup
```

Comment:
Can we give this a better name? As far as I can tell it's used to add stubbing capability to places in test files that have some tests in them (contexts, describes). `ComponentStubHelper`?

Reply:
One might argue that this file defined the entire `type: :view_component` type of specs which are usually addressed by ExampleGroup (like "RSpec::Rails::ModelExampleGroup".) This particular implementation also relies on existence of defined `component` which might not be required for other contexts stubbing might be required (like view/request). Maybe this module should be simplified then (without component) and we could create `ViewComponentExampleGroup` then outside of view_component folder?

Reply:
I actually like the idea of stubbing components in a views a lot, but we can probably add that capability in a future PR (well, relatively close future I think, from the looks of the prior longer PR - I think about testing that a banner is shown in view in some scenarios).

Reply:
@kicferk1 - This is one thing that I keep thinking about actually. Who should be responsible to make the decision whether given component should render or not? It looks like it lies somewhere between parent and the component itself - on one hand I'd like to test in the view that banner under certain conditions, but it is a banner component that knows what is required for that component to be rendered.

ViewComponent allows to define `render?` method that stops the component from rendering. So, given ChallengedBanner component - it knows it makes no sense to be rendered when given partnership is not challenged, so it seems it should go into its `render?` logic. But this leaves us with unconditional component rendering in the parent view `<%= render ChallengeBanner.new(parnership: partnership) %>` which feels a bit odd? Thoughts on that?

Reply:
This is probably something to discuss in greater detail on the PR adding the banner, but I strongly believe it should be the parent's responsibility to decide whether to render a component. 

That's because if it's the child's, and after you get a page with several components, then the logic of what gets rendered on a page gets split out across many files and possibly duplicated. Which makes it harder to read in the parent, harder to refactor, etc.

It gets even harder to understand what will render in a particular scenario when you get nested components.

Now, there are cases when putting that logic in the parent is just not a good idea - when it is complex and depends on internal component state. That's imo the time to shine for `render?`.